src/Adapter/MSTest.CoreAdapter/Execution/TypeCache.cs

```diff
@@ -308,6 +308,24 @@ private TestClassInfo CreateClassInfo(Type classType, TestMethod testMethod)
                         // Update test initialize/cleanup method from base type.
                         this.UpdateInfoIfTestInitializeOrCleanupMethod(classInfo, methodInfo, true, instanceMethods, testInitializeAttributeType, testCleanupAttributeType);
                     }
+
+                    if (this.IsAssemblyOrClassInitializeMethod(methodInfo, classInitializeAttributeType))
```

Comment:
similar to the above check, maybe we could limit this only to public and static methods.

Reply:
I was leveraging the use of the `methodInfo ` extension method (`HasCorrectClassOrAssemblyInitializeSignature`) inside the `this.IsAssemblyOrClassInitilizeMethod()`

It's already doing the check there to see if the method satisfies the right signature. So, in that case it would be a duplicated check for static and public. What do you think?

Reply:
well that first seems to be checking if the Assembly/ClassInitialize attribute is defined on the method. Attribute retrieval is generally costly and should be avoided if we can.

Reply:
hmm, that's true. I've changed this and also in previous lines when checking for init/cleanup in derived class. And, since this was similar in both derived class init/cleanup method and base, I extracted a method similar to `UpdateInfoIfTestInitializeOrCleanupMethod` but for class initialize/cleanup. That way we check for the class initialize/cleanup attribute only if the method is public and static.

I'll be pushing those changes after other comments are resolved.